src/components/SearchBox.tsx

```diff
@@ -201,7 +201,7 @@ export default class SearchBox<OptionType> extends React.PureComponent<
           })}
 
           {!extraActionElement &&
-            !results.length &&
+            (!results.length || !results[0]) &&
```

Comment:
dont add usecase specific code to library. pass results prop as empty array when "no results" state is expected